src/alarm/AlarmList.tsx

```diff
@@ -0,0 +1,112 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import AlarmListItem from './AlarmListItem';
+
+interface IComment {
+  comment: string;
+}
+
+interface IAlarm {
+  seen: boolean;
+  _id: string;
+  user: string;
+  comment: IComment;
+  post: string;
+}
+
+const TOKEN_KEY = 'token';
+
+const AlarmList = () => {
+  const [alarmList, setAlarmList] = useState<IAlarm[]>();
+  const [alramStatus, setAlramStatus] = useState<boolean | undefined>(true);
+  const [showedAlarm, setShowedAlarm] = useState(false);
+
+  // by 민형, 이미 모든 알람을 열어본 경우에는 해당 로직이 수행되지 않도록(리팩토링)_230111
+  const confirmAlarmList = async () => {
+    const item = localStorage.getItem(TOKEN_KEY);
+    const storedValue = item ? JSON.parse(item) : '';
+    await axios
+      .put(
+        'http://kdt.frontend.3rd.programmers.co.kr:5006/notifications/seen',
+        {},
+        {
+          headers: { Authorization: `bearer ${storedValue}` },
+        }
+      )
+      .then((res) => {
+        fetchAlarmData();
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
+  const fetchAlarmData = async () => {
+    const item = localStorage.getItem(TOKEN_KEY);
+    const storedValue = item ? JSON.parse(item) : '';
+
+    await axios
+      .get('http://kdt.frontend.3rd.programmers.co.kr:5006/notifications', {
+        headers: { Authorization: `bearer ${storedValue}` },
+      })
+      .then((res) => {
+        setAlarmList(res.data);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  };
+
+  const viewAlarmList = async () => {
+    if (showedAlarm) {
+      setShowedAlarm(false);
+      return;
+    }
+
+    setShowedAlarm(true);
```

Comment:
```suggestion
    setShowedAlarm(showedAlarm ? false : true);
```

로 바꾸셔도 될 거 같아요~
showedAlarm 는 모달 토글을 나타내는 건가요?? 그렇다면 네이밍을 변경하시는 게 좋을 거 같아요!